src/rune-bridge/src/rune_info.rs

```diff
@@ -0,0 +1,106 @@
+use candid::types::{Serializer, Type};
+use candid::{CandidType, Deserialize};
+use ordinals::{Rune, RuneId};
+use serde::Deserializer;
+use std::fmt::{Display, Formatter};
+use std::hash::{Hash, Hasher};
+use std::str::FromStr;
+
+#[derive(Debug, Copy, Clone, CandidType, Deserialize)]
+pub struct RuneInfo {
+    pub name: RuneName,
+    pub decimals: u8,
+    pub block: u64,
+    pub tx: u32,
+}
+
+impl RuneInfo {
+    pub fn id(&self) -> RuneId {
+        RuneId {
+            block: self.block,
+            tx: self.tx,
+        }
+    }
+
+    pub fn name(&self) -> RuneName {
+        self.name
+    }
+
+    pub fn name_array(&self) -> [u8; 32] {
+        let mut value = [0; 32];
+        let name = self.name.to_string();
+        let name_bytes = name.as_bytes();
+        let len = name_bytes.len().min(32);
+        value[0..len].copy_from_slice(&name_bytes[0..len]);
+        value
+    }
+
+    pub fn symbol_array(&self) -> [u8; 16] {
+        let mut value = [0; 16];
+        let name = self.name.to_string();
+        let name_bytes = name.as_bytes();
+        let len = name_bytes.len().min(16);
+        value[0..len].copy_from_slice(&name_bytes[0..len]);
+        value
+    }
+
+    pub fn decimals(&self) -> u8 {
+        self.decimals
+    }
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct RuneName(Rune);
+
+impl RuneName {
+    pub fn from_str(name: &str) -> Result<Self, <Rune as FromStr>::Err> {
```

Comment:
It has clippy warning `method `from_str` can be confused...` Why can't we implement From<&str> trait as opposed to this ?